sentry-types/src/protocol/envelope.rs

```diff
@@ -265,53 +303,59 @@ impl Envelope {
     where
         W: Write,
     {
-        let mut item_buf = Vec::new();
-
         // write the headers:
         let event_id = self.uuid();
         match event_id {
             Some(uuid) => writeln!(writer, r#"{{"event_id":"{uuid}"}}"#)?,
             _ => writeln!(writer, "{{}}")?,
         }
 
-        // write each item:
-        for item in &self.items {
-            // we write them to a temporary buffer first, since we need their length
-            match item {
-                EnvelopeItem::Event(event) => serde_json::to_writer(&mut item_buf, event)?,
-                EnvelopeItem::SessionUpdate(session) => {
-                    serde_json::to_writer(&mut item_buf, session)?
-                }
-                EnvelopeItem::SessionAggregates(aggregates) => {
-                    serde_json::to_writer(&mut item_buf, aggregates)?
-                }
-                EnvelopeItem::Transaction(transaction) => {
-                    serde_json::to_writer(&mut item_buf, transaction)?
-                }
-                EnvelopeItem::Attachment(attachment) => {
-                    attachment.to_writer(&mut writer)?;
+        match &self.items {
+            Items::Raw(bytes) => writer.write_all(bytes)?,
+            Items::EnvelopeItems(items) => {
```

Comment:
```suggestion
        let items = match &self.items {
            Items::Raw(bytes) => return writer.write_all(bytes).map(|_| ()),
            Items::EnvelopeItems(items) => items
```

This could reduce the excessive rightward drift.